src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -1229,6 +1231,24 @@ bool QueryAnalyzer::isTableExpressionNodeType(QueryTreeNodeType node_type)
         node_type == QueryTreeNodeType::QUERY || node_type == QueryTreeNodeType::UNION;
 }
 
+DataTypePtr QueryAnalyzer::getExpressionNodeResultTypeOrNull(const QueryTreeNodePtr & query_tree_node)
+{
+    auto node_type = query_tree_node->getNodeType();
+
+    if (node_type == QueryTreeNodeType::COLUMN || node_type == QueryTreeNodeType::CONSTANT)
```

Comment:
Better to use switch to allow the compiler to perform more aggressive optimizing.

Reply:
Updated.